cpp/dolfin/fem/ReferenceCellTopology.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2019 Chris Richardson and Garth N. Wells
+//
+// This file is part of DOLFIN (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#pragma once
+
+#include <array>
+#include <map>
+#include <set>
+
+namespace dolfin
+{
+
+enum class CellType : short int
+{
+  point,
+  interval,
+  triangle,
+  quadrilateral,
+  tetrahedron,
+  hexahedron
+};
+
+namespace fem
+{
+
+class ReferenceCellTopology
+{
+public:
+  static int dim(CellType cell_type);
+  static const int* num_entities(CellType cell_type);
+  static int num_vertices(CellType cell_type);
```

Comment:
We don't need `num_vertices`, `num_edges` and `num_faces`, can just use `num_entities()[0]` etc.